presto-main/src/main/java/com/facebook/presto/sql/planner/PlanFragmenter.java

```diff
@@ -699,6 +700,11 @@ public FragmentProperties(PartitioningScheme partitioningScheme)
             return children;
         }
 
+        public Set<PlanFragmentId> getDependencies()
```

Comment:
Now `children` and `dependencies` are interesting:
```
- `children` is `SubPlan`, and (when executes) are connected by remote exchange.
- `dependencies` is `PlanFragment`, and (when executes) are connected by TableWrite + TableScan.
```

This is not so easy to understand at first glance, since there are two types of "dependency" and they are tracked by two different classes. Can they be both tracked by `SubPlan` ? :)  I will also tries to read the later commits. 

BTW: Is there any reason to use `PlanFragmentId` instead of `PlanFragment` here ?

Reply:
`PlanFragment` is not a tree node. `SubPlan` is. And it looks like linking together `PlanFragment`'s into a tree is the only responsibility of the `SubPlan` class. `PlanFragment` defines relations with other fragments already, see: `PlanFragment#getRemoteSourceNodes`. The `RemoteSourceNode` contains the `List<PlanFragmentId> sourceFragmentIds`.